### PR TITLE
Add user settings for warn before exit

### PR DESCRIPTION
**Pull request by gsouquet**
Fixes vector-im/element-web#16620

Related pull requests
* vector-im/element-desktop#173
* vector-im/element-web#16781

### CODE REVIEW

File: src/components/views/settings/tabs/user/PreferencesUserSettingsTab.js [hunk 1]
```diff
@@ -161,6 +175,14 @@ export default class PreferencesUserSettingsTab extends React.Component {
...
         }
 
+        let warnBeforeExitOption = null;
+        if (this.state.warnBeforeExitSupported) {
+            warnBeforeExitOption = <LabelledToggleSwitch
+                value={this.state.warnBeforeExit}
+                onChange={this._onWarnBeforeExitChange}
+                label={_t('Warn before quitting')} />;
+        }
+
```
Comment (turt2live) [inline src/components/views/settings/tabs/user/PreferencesUserSettingsTab.js:185]: I think these sorts of settings are what Design would typically want to take a look at, given we try to not have too many options and instead try to design things that make sense instead.

I can see this being potentially controversial to turn on, so a setting would be appreciated by me at least.

**Review by turt2live — APPROVED** [review]
looks good otherwise

File: src/components/views/settings/tabs/user/PreferencesUserSettingsTab.js (hunk 1 above)
Comment (gsouquet) [reply src/components/views/settings/tabs/user/PreferencesUserSettingsTab.js:185]: I have run that through @nadonomy who suggested to add that option in "Settings > Preferences > General"

I believe the copy is very much up for debate at this point

**Comment by jryans** [comment]
We reviewed this with Nad over a video call.

**Pull request merged**